doc/user/content/sql/system-catalog/mz_internal.md

```diff
@@ -198,6 +198,35 @@ system.
 
+<!-- ### `mz_prepared_statement_history` -->
+
+<!-- The `mz_prepared_statement_history` table contains a subset of all -->
+<!-- statements that have been prepared. It only contains statements that -->
+<!-- have one or more corresponding executions in inclusion in -->
+<!-- [`mz_statement_execution_history`](#mz_statement_execution_history). -->
+
+<!-- | Field         | Type                         | Meaning                                                                                                                           | -->
+<!-- |---------------|------------------------------|-----------------------------------------------------------------------------------------------------------------------------------| -->
+<!-- | `id`          | [`uuid`]                     | The globally ID of the prepared statement.                                                                                        | -->
+<!-- | `session_id`  | [`uuid`]                     | The globally unique ID of the session that prepared the statement. Corresponds to [`mz_session_history.id`](#mz_session_history). | -->
+<!-- | `name`        | [`text`]                     | The name of the prepared statement (the default prepared statement's name is the empty string).                                   | -->
+<!-- | `sql`         | [`text`]                     | The SQL text of the prepared statement.                                                                                           | -->
+<!-- | `prepared_at` | [`timestamp with time zone`] | The time at which the statement was prepared.                                                                                     | -->
+
+<!-- ### `mz_session_history` -->
+
+<!-- The `mz_session_history` table contains all the sessions that have -->
+<!-- been established in the last 30 days, or (even if older) that are -->
+<!-- referenced from -->
+<!-- [`mz_prepared_statement_history`](#mz_prepared_statement_history). -->
+
+<!-- | Field          | Type                         | Meaning                                                                                                                           | -->
+<!-- |----------------|------------------------------|-----------------------------------------------------------------------------------------------------------------------------------| -->
+<!-- | `id`           | [`uuid`]                     | The globally unique ID of this history entry. Does **not** correspond to [`mz_sessions.id`](#mz_sessions), which can be recycled. | -->
+<!-- | `connected_at` | [`timestamp with time zone`] | The time at which the session was established.                                                                                    | -->
+<!-- | `application`  | [`text`]                     | The `application` session metadata field.                                                                                         | -->
```

Comment:
> If so, can you include details on how we handle a user changing this value mid-session?

That is a good question which I hadn't considered. I assumed this value was set during session startup and was immutable, but now that I'm looking at the code it seems that was wrong, and it's controlled by a var.

The main reason we want this is to distinguish between different surfaces: psql, mz, the web console, the web shell, etc. Do you know if those are likely to be available on session startup, or do they only get set later? I can also check this myself if you don't know.

Reply:
This is a recurring unsolved issue. The application name is a session variable. Like all session variables it has the following properties: 

- It can be initialized via the connection string.
  - If it's not initialized via the connection string, then it's initialized to some default value, in the case of `application_name` the default value is the empty string.
- It can be modified at any time via `SET application_name ...`.
- Changing the value is done transitionally, i.e. if the value is set in a transaction and the transaction aborts, then the change is rolled back.
- The value can be changed temporarily for an ongoing transaction. i.e. `SET LOCAL application_name ...` will change the application name for the duration of the transaction, then roll it back when the transaction commits.

Given all of this, it's unclear how `application_name` or any other session variable should be written down in a catalog table.

One idea was that we could write down the application name that appeared in the connection string, if any, and indicate in the docs that it may not be the current value of the application name. I think this got shot down, but I can't remember why.

Reply:
This is really annoying because some drivers might run this as their first command, but that's after startup has occurred. Regardless, we should start with the value after startup (which will reflect anything in the connection string, including our web console). If that ends up not being enough we can do more.

Reply:
I think being able to distinguish between psql, mz, web console, and web shell will give us 90% of the possible value here.